### ISSUE TITLE
motif mapping not correct?

### ISSUE THREAD

**Issue by YichaoOU** [jhkorhonen/MOODS#26]
### Input: test.fa

> \>test
A**R**G**R**GGAAGTGAATG
\>test2
AAGAGGAAGTGAATG

### Input: motif matrix (PWM)

| 0.562501 | 0.461538 | 0.206731 | 0.576922 | 0.139423 | 0.072115 | 0.985577 | 0.975962 | 0.149038 | 0.048077 | 0.028846 | 0.817307 | 0.572115 | 0.711539 | 0.269231 |
|----------|----------|----------|----------|----------|----------|----------|----------|----------|----------|----------|----------|----------|----------|----------|
| 0.086538 | 0.033654 | 0.139423 | 0.067308 | 0.0      | 0.0      | 0.0      | 0.0      | 0.365385 | 0.024038 | 0.0      | 0.004808 | 0.091346 | 0.091346 | 0.350962 |
| 0.182692 | 0.235577 | 0.600961 | 0.302885 | 0.798077 | 0.913462 | 0.0      | 0.0      | 0.485577 | 0.024038 | 0.971154 | 0.177885 | 0.269231 | 0.076923 | 0.283654 |
| 0.168269 | 0.269231 | 0.052885 | 0.052885 | 0.0625   | 0.014423 | 0.014423 | 0.024038 | 0.0      | 0.903847 | 0.0      | 0.0      | 0.067308 | 0.120192 | 0.096154 |

### command 

`python ../moods_dna.py -S M4453_1.02_BCL11A.pfm  -s test.fa -t 0`

### output

>test,M4453_1.02_BCL11A.pfm,0,+,9.802885,ARGRGGAAGTGAATG,aGgAggaagtgaatg
test,M4453_1.02_BCL11A.pfm,0,+,9.528848,ARGRGGAAGTGAATG,aGgGggaagtgaatg
test,M4453_1.02_BCL11A.pfm,0,-,1.052883,ARGRGGAAGTGAATG,aGgAggaagtgaatg
test,M4453_1.02_BCL11A.pfm,0,-,1.057691,ARGRGGAAGTGAATG,aGgGggaagtgaatg
test2,M4453_1.02_BCL11A.pfm,0,+,10.028846,AAGAGGAAGTGAATG,
test2,M4453_1.02_BCL11A.pfm,0,-,1.081729,AAGAGGAAGTGAATG,

### question

MOODS is supposed to give me all combinations of the input SNPs, right?

Clearly, the 5th row, where the score is 10, is not captured in `test` sequence. 

So there is a bug somewhere in MOODS?

Thanks,
Yichao

**Comment by jhkorhonen** [comment]
Looks like a bug. Thanks, I'll have to figure this out.

**Comment by jhkorhonen** [comment]
Fixed in 19e9e11.

**Comment by YichaoOU** [comment]
Hi, 

It did work for the example I gave but seems it doesn't work for more general sequences, such as `ARRRRGAAGTGAATG`

**Comment by YichaoOU** [comment]
OK, found the bug, if I include more than 2 SNPs in one TFBS, then your program won't work.

**Comment by jhkorhonen** [comment]
Right, I'm now somewhat amazed that I've never managed to hit any of these cases myself earlier. More fixes in 90cfb54; I'll do more thorough testing in the near future.

**Comment by YichaoOU** [comment]
Thank you so much! As far as I can tell, my problem is fixed.
